### DISCUSSION TITLE
The Proper Way for eGPU is Thunderbolt 2/3/4

### DISCUSSION THREAD

**Discussion by juanpc2018** [geerlingguy/raspberry-pi-pcie-devices#591] [General]
https://egpu.io/best-egpu-buyers-guide/

i have a [Razer Core X Chroma](https://www.razer.com/gb-en/gaming-egpus/Razer-Core-X/RC21-01430100-R3W1), [modified](https://egpu.io/forums/thunderbolt-enclosures/razer-core-x-chroma-unbiased-review-from-an-owner/) with 140mm front fan + small nylon cable ties, and Noctua 80mm PSU fan.
Recommended.
Razer X Core Not Chroma is a bit shorter, does Not have RGB LEDs, No USB3, No Ethernet, PSU is -50w smaller.

MacMini 2014 has TB2,
most eGPU are TB3, requires Bi-directional Apple Thunderbolt adapter + TB2 cable.
for M.2 PCIe to Thunderbolt2, adds a weird latency when used as boot drive, better boot from USB3.
for graphics work ok,
but is better TB3 to TB3 / 4 directly.

OSX removed support for TB2 since OSX HighSierra 10.13.4, and added TB3.
there is a script that re-enables TB2 in OSX but... thats a Red Flag.
OSX Catalina 10.15.7 is the last 100% amd64,
OSX HighSierra 10.13.6 is the last x86_64,
OSX Mojave 10.14 removes 32-bit driver support, 
OSX BigSur 11 is hybrid Kernel,  memory runs -30% slower, there is No point installing hybrid OSX on intel, takes 4 ever to upgrade and cannot be undone.

MacMini 2018 has TB3 but also has T2 chip. stops installing Linux, unless its a special version compatible with T2.
another Red Flag.

8GB is the minimum for browsing internet.

OSX has very limited Graphics drivers,
NVIDIA Titan 6GB had glitches,
OSX is better with AMD, RX 290x 480 580 in PCIe, 
but Pulse, Nitro "maybe".
also requires to be Flashed to have UEFI Boot Menu "Apple Logo"
[macvidcards.com](http://www.macvidcards.com/)

i had HD7950 with UEFI and worked Ok with MacPro 5.1 2010 tower,
did Not work eGPU with MacMini 2014 and OSX 10.13.3
works PCIe upto Catalina, but does Not work eGPU in OSX,
another Red Flag.
seems NVIDIA are better for eGPU in OSX, but sold the Titan 6GB before i could test.

is a guessing game.

im using Kubuntu 22.04.* LTS dual kernel

Apple M3 Max is Amazing CPU [single core](https://www.cpu-monkey.com/en/cpu_benchmark-cinebench_2024_single_core)
but... the only Linux compatible with M1 M2 is [AsahiLinux](https://asahilinux.org/), [Remix](https://asahilinux.org/fedora/)
No M3 support yet, ¿could be M3 is [RISC-V](https://www.techpowerup.com/298936/report-apple-to-move-a-part-of-its-embedded-cores-to-risc-v-stepping-away-from-arm-isa) Not ARM ?

im interested in the Pi5 but Not having TB3 is a Red Flag.
Pi5 could work for PiStorm32 Motorola 68040 BareMetal CPU Emulation.
interesting that has UEFI.

ARMv8, Ready-to-Run:
[NVIDIA AGX Orin Dev Kit](https://developer.nvidia.com/buy-jetson) MiniBox
[NVIDIA IGX Orin](https://www.nvidia.com/en-us/edge-computing/products/igx/) ITX Board
Ampere GPU, Ubuntu 20.04 LTS,
ARMv8 CPU but $$$
there are other ARM options, No GPU, with PCIe.

[QEMU](https://www.qemu.org/download/#source) + [WineHQ](https://wiki.winehq.org/Download) for x86_64 apps on ARMv8 CPU could be an option, untested.

if there is a RISC-V with Thunderbolt3, Pi5 is DOA.

**Comment by geerlingguy** [comment]
Thunderbolt has it's own weird quirks, when it comes to PCIe delivery over TB3/TB4... but the bigger issue is ThunderBolt is not nearly as low-cost to implement as would be required for inclusion on devices sub-$99 (much less sub-$50, which is probably the best price target for Pi... hopefully a Pi 5 2 GB comes out at $40, I can dream...).

**Comment by juanpc2018** [comment]
Pi5 is tempting, but i think ill wait for the [P550](https://www.sifive.com/boards/hifive-pro-p550)

**Reply by geerlingguy** [reply]
That board doesn't seem like it'll have Thunderbolt either, though? But it is certainly an interesting dev board!

**Comment by OwnedByWuigi** [comment]
How did you get TB2 working? I'm on the "can't innovate anymore my ass" Mac Pro (late 2013) and need a eGPU enclosure that support TB2
